packages/ckeditor5-fullscreen/src/handlers/abstracteditorhandler.ts

```diff
@@ -806,4 +904,179 @@ export class FullscreenAbstractEditorHandler {
+				annotationsFilters.set( uiName, data.filter );
+			}
+		}
+
+		// First, check if someone has a filter defined for requested UI. If so, retrieve and apply it in fullscreen.
+		if ( annotationsFilters.has( uiName ) ) {
+			annotationsUIs.activate( uiName, annotationsFilters.get( uiName ) );
+		}
+		// If no filter is defined for requested UI, read the filters for the active display(s) mode and apply them.
+		// It's possible there are filters for modes other than selected, so display annotations that match any of them.
+		else if ( annotationsFilters.size ) {
+			annotationsUIs.activate( uiName,
+				( annotation: any ) => [ ...annotationsFilters.values() ].some( filter => filter( annotation ) )
+			);
+		}
+		// If no filters are defined for the active display mode(s), simply display all annotations in the requested UI.
+		else {
+			annotationsUIs.switchTo( uiName );
+		}
+	}
+
+	/**
+	 * Sets up a resize observer to adjust the visibility of the left and right sidebars dynamically.
+	 */
+	private _setupResizeObserver(): void {
+		const wrapper = this._wrapper!.querySelector( '.ck-fullscreen__editable-wrapper' ) as HTMLElement;
+
+		if ( this._resizeObserver ) {
+			this._resizeObserver.destroy();
+		}
+
+		this._resizeObserver = new ResizeObserver( wrapper, () => {
+			this._adjustVisibleElements();
+		} );
+	}
+
+	/**
+	 * Hides the left sidebar. Works only if there is anything to hide.
+	 */
+	// Code coverage is provided in the commercial package repository as integration unit tests.
+	/* istanbul ignore next -- @preserve */
+	private _hideLeftSidebar() {
+		if ( this._collapseLeftSidebarButton ) {
+			const leftSidebar = this._wrapper!.querySelector( '.ck-fullscreen__left-sidebar' ) as HTMLElement;
+
+			leftSidebar.classList.add( 'ck-fullscreen__left-sidebar--collapsed' );
+			this._collapseLeftSidebarButton.icon = IconDocumentOutlineToggle;
+			this._collapseLeftSidebarButton.tooltip = 'Show left sidebar';
+		}
+	}
+
+	/**
+	 * Shows the left sidebar. Works only if there is anything to show.
+	 */
+	// Code coverage is provided in the commercial package repository as integration unit tests.
+	/* istanbul ignore next -- @preserve */
+	private _showLeftSidebar() {
+		if ( this._collapseLeftSidebarButton ) {
+			const leftSidebar = this._wrapper!.querySelector( '.ck-fullscreen__left-sidebar' ) as HTMLElement;
+
+			leftSidebar.classList.remove( 'ck-fullscreen__left-sidebar--collapsed' );
+			this._collapseLeftSidebarButton.icon = IconPreviousArrow;
+			this._collapseLeftSidebarButton.tooltip = 'Hide left sidebar';
+		}
+	}
+
+	/**
+	 * Hides the right sidebar. Works only if there is anything to hide.
+	 */
+	private _hideRightSidebar() {
+		if ( this._wrapper!.querySelector( '.ck-fullscreen__right-sidebar' )!.firstChild ) {
+			this._switchAnnotationsUI( 'narrowSidebar' );
+			this._wrapper!.querySelector( '.ck-fullscreen__right-sidebar' )!.classList.add( 'ck-fullscreen__right-sidebar--collapsed' );
```

Comment:
The **whole** plugin would greatly benefit from using UI library abstraction. Right now, it's a spaghetti of `querySelectors`  that point to undocumented elements in DOM and `classList` manipulations that deal with arbitrary strings at certain moments. This makes the code really hard to read, understand, and maintain.